examples/libh2o/http1client.c

```diff
@@ -39,64 +39,59 @@ static h2o_iovec_t iov_filler;
 static int delay_interval_ms = 0;
 static int cur_body_size;
 
-static h2o_http1client_head_cb on_connect(h2o_http1client_t *client, const char *errstr, h2o_iovec_t **reqbufs, size_t *reqbufcnt,
-                                          int *method_is_head, h2o_http1client_proceed_req_cb *proceed_req_cb,
-                                          h2o_iovec_t *cur_body, int *body_is_chunked, h2o_url_t *origin);
-static h2o_http1client_body_cb on_head(h2o_http1client_t *client, const char *errstr, int minor_version, int status,
-                                       h2o_iovec_t msg, h2o_header_t *headers, size_t num_headers, int rlen);
+static h2o_httpclient_head_cb on_connect(h2o_httpclient_t *client, const char *errstr, h2o_iovec_t *method, h2o_url_t *url,
+                                         h2o_headers_t *headers, h2o_iovec_t *body, h2o_httpclient_proceed_req_cb *proceed_req_cb,
+                                         h2o_httpclient_features_t features, h2o_url_t *origin);
+static h2o_httpclient_body_cb on_head(h2o_httpclient_t *client, const char *errstr, int minor_version, int status, h2o_iovec_t msg,
+                                      h2o_header_t *headers, size_t num_headers, int rlen);
 
-static void start_request(h2o_http1client_ctx_t *ctx)
+static void start_request(h2o_httpclient_ctx_t *ctx)
 {
-    h2o_url_t url_parsed;
-    h2o_iovec_t *req;
+    h2o_url_t *url_parsed;
```

Comment:
> Is it due to an API change from http1client to httpclient, or is it due to an existing bug becoming apparent?

Since I wanted to store `url` in `client->data` for later use. Original code stored HTTP/1.1 request buffer, so this can be said as the change caused by API changes. Anyway this is an example, so I can change this easily if you like.

Reply:
Thank you for the clarification.

> Since I wanted to store `url` in `client->data` for later use.

I assume that you are passing `url` which is a pointer to `h2o_url_t *` as an argument to `h2o_httpclient_connect`. Assuming that that is true, I do not like this change, because it sounds like a silent API change where the lifetime of the argument that the caller needs to guarantee changes from "while the function is called" to sometime after than that. Generally speaking that is a dangerous change, because it is so easy to miss fixing existing code that relies on the current behavior.

Reply:
NOTE: As we discussed, This is not an API change, but the change caused by API changes. So this isn't dangerous and we left this as it is.